lib/matplotlib/collections.py

```diff
@@ -179,9 +181,30 @@ def get_offset_transform(self):
         return t
 
     def get_datalim(self, transData):
+
+        # Get the automatic datalim of the collection.
+        #
+        # This operation depends on the transforms for the data in the
+        # collection and whether the collection has offsets.
+        #
+        # 1) offsets = None, transform = transData: use the paths for the
+        # automatic limits (i.e. for LineCollection in streamline).
+        # 2) offsets != None: offset_transform = transData:
+        #    a) transform = transData: use the path + offset for limits (i.e
+        #    for bar).
+        #    b) transform != transData: just use the offsets for the limits
+        #    (i.e. for scatter)
+        # 3) otherwise return a null Bbox.
+
         transform = self.get_transform()
         transOffset = self.get_offset_transform()
+        if (not self._offsetsNone and
+            not transOffset.contains_branch(transData)):
```

Comment:
I guess(?) you could get a few extra points by checking `contains_branch_separately` and then deciding to use the `x` or the `y` data limits if one of them does contain transData; this could be a separate PR too as the use case is a bit obscure.

Reply:
let's just forget this case for now

Reply:
As discussed on gitter, I think thats best for simplicity sake.  For instance, I don't know what the datalim handling does with infs (for instance) and adding this case is a good chunk of code.  As also discussed, very happy to revisit if it turns out to be needed.